### PR TITLE
refactor(bazel): Create `ng-add` schematic for Bazel

### PR DESCRIPTION
**Pull request by kyliau**
The logic to create additional files needed for Bazel are currently
hosted in `ng new`. Such files include the `main.*.ts` files needed
for AOT and a different `angular.json` to use Bazel builder, among others.

This commit refactors the logic into `ng add` so that it can be used to
perform the same modifications in an existing project. Users could do so
by running `ng add @angular/bazel`.

With this change, `ng new` effectively becomes an orchestrator that runs
the original `ng new` followed by `ng add @angular/bazel`.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/bazel/src/schematics/bazel-workspace/index.ts [hunk 1]
```diff
@@ -72,25 +53,15 @@ export default function(options: BazelWorkspaceOptions): Rule {
...
+    if (!host.exists('yarn.lock')) {
+      host.create('yarn.lock', '');
+    }
 
     const workspaceVersions = {
       'RULES_NODEJS_VERSION': '0.16.5',
       'RULES_TYPESCRIPT_VERSION': '0.22.1',
-      'ANGULAR_VERSION': existingVersions.Angular || clean(latestVersions.Angular),
-      'RXJS_VERSION': existingVersions.RxJs || clean(latestVersions.RxJs),
+      'ANGULAR_VERSION': clean(latestVersions.Angular),
```
Comment (alexeagle) [inline packages/bazel/src/schematics/bazel-workspace/index.ts]: is this related to the change, or could it be a separate PR?
Just asking because it's easier to find a chunk of time that fits a smaller PR review

File: packages/bazel/src/schematics/ng-add/index.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    const angularCoreNode = findPropertyInAstObject(deps, '@angular/core');
+    if (!angularCoreNode) {
+      throw new Error('@angular/core dependency not found in package.json');
+    }
+    const angularCoreVersion = angularCoreNode.value as string;
+
+    const devDependencies: {[k: string]: string} = {
+      '@angular/bazel': angularCoreVersion,
+      // TODO(kyliau): Consider moving this to latest-versions.ts
+      '@bazel/bazel': '^0.22.0',
```
Comment (alexeagle) [inline packages/bazel/src/schematics/ng-add/index.ts]: maybe merge conflict with the other PR? should be 0.22.1?

File: packages/bazel/src/schematics/ng-add/index.ts [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+          },
+          'extract-i18n': {
+            'builder': '@angular-devkit/build-angular:extract-i18n',
+            'options': {'browserTarget': `${name}:build`}
+          },
+          'test': {
+            'builder': '@angular/bazel:build',
+            'options': {'bazelCommand': 'test', 'targetLabel': '//src/...'}
+          },
+          'lint': {
```
Comment (alexeagle) [inline packages/bazel/src/schematics/ng-add/index.ts]: do we really have to replace these keys that we don't care about?

File: packages/bazel/src/schematics/ng-add/index.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+ * Create a backup for the original angular.json file in case user wants to
+ * eject Bazel and revert to the original workflow.
+ */
+function backupAngularJson(): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    const workspacePath = getWorkspacePath(host);
+    if (!workspacePath) {
+      return;
+    }
+    host.create(`${workspacePath}.bak`, host.read(workspacePath));
```
Comment (alexeagle) [inline packages/bazel/src/schematics/ng-add/index.ts]: how will the user know we did this? should the .bak file have some explanatory comment added to the top? do we print something to the console during `ng add` that explains how to revert?

**Review by alexeagle — APPROVED** [review]

File: packages/bazel/src/schematics/bazel-workspace/index.ts (hunk 1 above)
Comment (kyliau) [reply packages/bazel/src/schematics/bazel-workspace/index.ts]: You're right, it's unrelated to this change. I should have resisted the temptation to change this file.
I reverted the unnecessary changes.

File: packages/bazel/src/schematics/ng-add/index.ts (hunk 3 above)
Comment (kyliau) [reply packages/bazel/src/schematics/ng-add/index.ts]: Changed code to leave lint and extract-i18n unchanged.

File: packages/bazel/src/schematics/ng-add/index.ts (hunk 4 above)
Comment (kyliau) [reply packages/bazel/src/schematics/ng-add/index.ts]: I was planning to have another schematic to help user revert to the original workflow. Regardless, it's good idea to add a note in the backup file so user would not accidentally delete this file.

File: packages/bazel/src/schematics/ng-add/index.ts (hunk 2 above)
Comment (kyliau) [reply packages/bazel/src/schematics/ng-add/index.ts]: Yes, it should be 0.22.1
